### PR TITLE
Add CPP tests for some group operations and add cpp to the build

### PR DESCRIPTION
**Pull request by coltfred**

### CODE REVIEW

File: cpp/test/test.cpp [hunk 1]
```diff
@@ -42,6 +48,30 @@ std::string vec_to_string(RustVeci8 a)
...
+    }
+
+    return random_string;
+}
+std::string random_id()
+{
+    return random_string(94);
+}
+
+auto deviceContextString = "{\"accountId\":\"abcABC012_.$#|@/:;=+'-91e078f0-a60c-4251-8652-dd498c07a8f4\",\"segmentId\":1825,\"signingPrivateKey\":\"uKHa70uwLVG3IU7XodT2kla/PuC/En8PkRCjMMc9ZE7HFrOV+g0vOwATp/CiXp65mVas0K6TSl/RaxDGlcmsnA==\",\"devicePrivateKey\":\"YZRlDSkM+JxxSXCtWCVK693qfhNqcbhaPrtHs92uD4w=\"}";
```
Comment (giarc3) [inline cpp/test/test.cpp:73]: Where is this device context from? We've recently been trying to be conscious about what devices we make public, so just checking

Comment (coltfred) [reply cpp/test/test.cpp:73]: I stole it from an ironoxide unit test. If you have the ability to generate one for the same segment we use for ironoxide-android, we should just use one of those instead.

File: cpp/test/test.cpp [hunk 2]
```diff
@@ -42,6 +48,30 @@ std::string vec_to_string(RustVeci8 a)
     return std::string(reinterpret_cast<char const *>(&a[0]), a.size());
 }
 
+std::string random_string(std::size_t length)
```
Comment (ernieturner) [inline cpp/test/test.cpp:51]: Just curious, did you figure out if this was actually random?

File: cpp/test/test.cpp [hunk 3]
```diff
@@ -64,17 +94,76 @@ void test_user_id_error(void)
...
+    auto group_name = unwrap(GroupName::validate("blargh"));
+    TEST_CHECK(group_name.getName().to_std_string() == "blargh");
+}
+
+void group_create_default(void)
+{
+    DeviceContext d = unwrap(DeviceContext::fromJsonString(deviceContextString));
+    IronOxide sdk = unwrap(IronOxide::initialize(d, IronOxideConfig()));
+    auto group_create_result = unwrap(sdk.groupCreate(GroupCreateOpts()));
+    TEST_CHECK(group_create_result.getMemberList().getList().as_slice().size() == 1);
```
Comment (ernieturner) [inline cpp/test/test.cpp:116]: Add a check for `getAdminList()` too maybe? Although I guess that's just getting down into testing the core of IronOxide itself, so maybe the goal is mostly just "did this pass without segfaulting and return an expected value".

File: cpp/test/test.cpp (hunk 2 above)
Comment (coltfred) [reply cpp/test/test.cpp:51]: If you look at the commit history you'll see that this is a different method (which is actually random). So in short, yes this is random.

File: cpp/test/test.cpp (hunk 3 above)
Comment (coltfred) [reply cpp/test/test.cpp:116]: This is basically sanity checking, ya. Not really interested in replicating a bunch of tests. Doing the calls is the important bit.

**Comment by coltfred** [comment]
I increased the things we assert about. We don't current support equality on our types #121 and we don't have jwt support. I wrote a ticket about bringing in jwt and finishing the tests.

**Pull request merged**